### ISSUE TITLE
Installation fails on Debian 10

### ISSUE THREAD

**Issue by JKatzwinkel** [JKamlah/ocrd_pagetopdf#17]
Make target `deps-ubuntu` fails on Debian 10 (buster), due to dependency `openjdk-8-jre-headless` not being in the standard packages.  Is `ocrd_pagetopdf` compatible with Java 11 and would it be possible to add to your Makefile a conditional along the lines of

```Makefile
ifeq ($(shell lsb_release -rs),10)
apt-get install -y openjdk-11-jre-headless
endif
```

...or something like this in order to have installation work on Debian 10?

**Comment by kba** [comment]
The question is whether https://github.com/PRImA-Research-Lab/prima-page-to-pdf is compatible with Java `!= 8`. From past experience with PRImA tools I doubt it, but please try it and find out for yourself with your proposed change.

**Comment by bertsky** [comment]
> The question is whether https://github.com/PRImA-Research-Lab/prima-page-to-pdf is compatible with Java `!= 8`. From past experience with PRImA tools I doubt it, but please try it and find out for yourself with your proposed change.

AFAICT all PRImA libs need to be reuilt for Java 11. It's possible, I've done it. But the whole Eclipse-based build is undocumented, and there are some problems with it – see discussion [here](https://github.com/PRImA-Research-Lab/prima-page-viewer/pull/18)

**Comment by JKatzwinkel** [comment]
Thanks for your quick and informative responses! I'm gonna put a note into [our documentation](https://github.com/StaatsbibliothekBerlin/ocrd_butler) suggesting some kind workaround. Feel free to close!

**Comment by stweil** [comment]
I just tried Java `openjdk version "16-ea" 2021-03-16` [sic] with Debian bullyeye. It created a PDF file and did not complain, so maybe it works better than expected with newer versions of Java.

**Comment by bertsky** [comment]
> I just tried Java `openjdk version "16-ea" 2021-03-16` [sic] with Debian bullyeye. It created a PDF file and did not complain, so maybe it works better than expected with newer versions of Java.

In that case, perhaps a neutral `default-jdk` would be the better dependency for the moment?

**Comment by stweil** [comment]
Yes, as soon as my preliminary result is confirmed. I got a PDF with the image, but no selectable text. That might be caused by my test data, but could also be caused by the Java version.

**Comment by stweil** [comment]
I had to add `-text-source W` when running `PageToPdf.jar`. Now the generated PDF has a working text layer.

So it seems to work fine.

**Comment by stweil** [comment]
@JKatzwinkel, @bertsky, I changed `Makefile` to install `default-jre-headless` now. That should work and fix this issue.

**Comment by JKatzwinkel** [comment]
Very nice, I seem to be able to `make all` of `ocrd_all` again. Thanks a ton!

**Comment by mikegerber** [comment]
> > The question is whether https://github.com/PRImA-Research-Lab/prima-page-to-pdf is compatible with Java `!= 8`. From past experience with PRImA tools I doubt it, but please try it and find out for yourself with your proposed change.
> 
> AFAICT all PRImA libs need to be reuilt for Java 11. It's possible, I've done it. 

Except for prima-page-converter's dependency PrimaText.jar, which is [mostly closed-source](https://github.com/PRImA-Research-Lab/prima-page-converter/issues/17).
